### PR TITLE
create_jenkins_plugin.md: Add goal and motivation

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <ce.gao@outlook.com>

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: data/projects/create_jenkins_plugin.md [hunk 1]
```diff
@@ -6,6 +6,20 @@ If you find this fun you'll be the one knowing this component best and will
...
+Jenkins is the most popular ci-cd server, and there are many corporations 
+who already use it. Integrating coala into Jenkins can attract business 
+clients better who could end up sponsoring/helping out coala.
+
+#### Final Goal
+
+The aim of the project is to make a plugin for Jenkins which can do 
+static analysis on code and make these visible to the user. coala 
+provides a framework for static analysis along with modularized 
+analyzers called bears. In the project, coala will be extended to
```
Comment (sils) [inline data/projects/create_jenkins_plugin.md]: hah: `coala provides a framework for static analysis along with modularized analyzers called bears.` why are you dumping a random coala description into the goal? :rofl:

Comment (gaocegege) [inline data/projects/create_jenkins_plugin.md]: I CP this from a former coala GSoC proposal 
😅

File: data/projects/create_jenkins_plugin.md [hunk 2]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
+
+#### Final Goal
+
+The aim of the project is to make a plugin for Jenkins which can do 
+static analysis on code and make these visible to the user.
+
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### GSOC 2017 COMMUNITY BONDING(May 4 - May 30)
```
Comment (yukiisbored) [inline data/projects/create_jenkins_plugin.md]: Please add spacing before the parentheses

File: data/projects/create_jenkins_plugin.md [hunk 3]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
 
 * A test jenkins instance exists.
 * An official repository exists with rultor set up.
 
-##### GSOC 2017 MIDTERM
+##### GSOC 2017 PHASE 1(May 30 - June 30)
```
Comment (yukiisbored) [inline data/projects/create_jenkins_plugin.md]: Please add spacing before the parentheses

File: data/projects/create_jenkins_plugin.md [hunk 4]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
+
+* A parser exists for parsing coala result.
+
+##### GSOC 2017 PHASE 2(June 30 - July 28)
```
Comment (yukiisbored) [inline data/projects/create_jenkins_plugin.md]: Please add spacing before the parentheses

File: data/projects/create_jenkins_plugin.md [hunk 5]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
 
 * A working plugin with minimal functionality exists.
 * A basic testsuite exists.
 * Basic documentation exists.
 
-##### GSOC 2017 FINAL
+##### GSOC 2017 FINAL(July 28 - August 29)
```
Comment (yukiisbored) [inline data/projects/create_jenkins_plugin.md]: Please add spacing before the parentheses

File: data/projects/create_jenkins_plugin.md [hunk 6]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
 get the chance to maintain it on your own if you want - always with the
 support of the community guarding your back of course.
 
+#### Motivation
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: Just put the motivation parahraph at the top without any header.

File: data/projects/create_jenkins_plugin.md [hunk 7]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
+
+Jenkins is the most popular ci-cd server, and there are many corporations 
+who already use it. Integrating coala into Jenkins can attract business 
+clients better who could end up sponsoring/helping out coala.
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: It also generally allows a wider range of users to use coala in their prefered way.

File: data/projects/create_jenkins_plugin.md [hunk 8]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
 get the chance to maintain it on your own if you want - always with the
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: i'd remove this paragraph. Instead add some details on how such plugin could be built (I have no idea but maybe using the json output for example). Anything else that could be helpful or interesting for the student also should be added.

File: data/projects/create_jenkins_plugin.md [hunk 9]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
+#### Motivation
+
+Jenkins is the most popular ci-cd server, and there are many corporations 
+who already use it. Integrating coala into Jenkins can attract business 
+clients better who could end up sponsoring/helping out coala.
+
+#### Final Goal
+
+The aim of the project is to make a plugin for Jenkins which can do 
+static analysis on code and make these visible to the user.
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: Maybe:
"The goal of this project is to offer a coala plugin for Jenkins to allow easy integration of coala and thus making it easier for some developers to use it. Another benefit could be increased viability for companies to use coala."

File: data/projects/create_jenkins_plugin.md [hunk 10]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
+
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### GSOC 2017 COMMUNITY BONDING(May 4 - May 30)
 
 * A test jenkins instance exists.
 * An official repository exists with rultor set up.
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: CI should exist. Rultor is not that important and rather easy.

File: data/projects/create_jenkins_plugin.md [hunk 11]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
 
-##### GSOC 2017 MIDTERM
+##### GSOC 2017 PHASE 1(May 30 - June 30)
+
+* A parser exists for parsing coala result.
+
+##### GSOC 2017 PHASE 2(June 30 - July 28)
 
 * A working plugin with minimal functionality exists.
 * A basic testsuite exists.
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: The testsuite should test everything that exsits at that point. As the plugin is just rudementary, the testsuite also is simple at this stage.

File: data/projects/create_jenkins_plugin.md [hunk 12]
```diff
@@ -6,20 +6,35 @@ If you find this fun you'll be the one knowing this component best and will
...
 * Basic documentation exists.
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: Same as for tests

File: data/projects/create_jenkins_plugin.md [hunk 13]
```diff
@@ -1,26 +1,45 @@
...
+could help you to build the plugin. 
+
+There is a similiar plugin in Jenkins plugin repository, 
+[warnings plugin](https://wiki.jenkins-ci.org/display/JENKINS/Warnings+Plugin). 
+A parser for warnings plugin could be implemented in the early 
+stage of developing. It's easy to get started.
+
+#### Final Goal
+
+The goal of this project is to offer a coala plugin for Jenkins to
```
Comment (adhikasp) [inline data/projects/create_jenkins_plugin.md]: Could this goal explain further about what a coala jenkin plugin could do? Is it to make installing coala and it's configuration easier in jenkin workflow then could output the result in nice way inside jenkin console?

File: data/projects/create_jenkins_plugin.md [hunk 14]
```diff
@@ -1,26 +1,45 @@
...
+There is a similiar plugin in Jenkins plugin repository, 
+[warnings plugin](https://wiki.jenkins-ci.org/display/JENKINS/Warnings+Plugin). 
+A parser for warnings plugin could be implemented in the early 
+stage of developing. It's easy to get started.
+
+#### Final Goal
+
+The goal of this project is to offer a coala plugin for Jenkins to 
+allow easy integration of coala and thus making it easier for some 
+developers to use it. Another benefit could be increased viability for
```
Comment (adhikasp) [inline data/projects/create_jenkins_plugin.md]: This benefit to company is already stated above. I dont think it's necessary to say it again.

File: data/projects/create_jenkins_plugin.md [hunk 15]
```diff
@@ -1,26 +1,45 @@
-Jenkins users should be able to see coala results right in the 
-Jenkins UI. The result of this project should be a full fledged Jenkins 
-integration, living in an own repository with a full testsuite.
+Jenkins is the most popular ci-cd server, and there are many corporations 
+who already use it. Integrating coala into Jenkins can attract business 
+clients better who could end up sponsoring/helping out coala. It also generally allows a wider range of users to use coala in their prefered way.
```
Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: linebreaks :D

Comment (sims1253) [inline data/projects/create_jenkins_plugin.md]: Also what problem does this project try to fix? On conferences we encounter numerous devs that ask for a jenkins plugin but can't use coala without one.

Comment (gaocegege) [inline data/projects/create_jenkins_plugin.md]: In my opinion, now we could use coala in jenkins by writing shell commands to install and run coala, and see result in build logs.

And this plugin allows users to run coala from jenkins UI, and see the result just like warnings plugin.

I get the idea from docker plugin. Users could run docker command from shell without this plugin but if they install this plugin, the process is clear.

File: data/projects/create_jenkins_plugin.md (hunk 13 above)
Comment (gaocegege) [inline data/projects/create_jenkins_plugin.md]: ref https://github.com/coala/projects/pull/186#discussion_r99809577

File: data/projects/create_jenkins_plugin.md (hunk 14 above)
Comment (gaocegege) [inline data/projects/create_jenkins_plugin.md]: OK

File: data/projects/create_jenkins_plugin.md (hunk 15 above)
Comment (gaocegege) [inline data/projects/create_jenkins_plugin.md]: WDYT @adhikasp

**Comment by sims1253** [comment]
ack cc7d5831f1dd266add40026fbcab35764a0b3697

**Comment by sims1253** [comment]
@rultor merge

**Pull request merged**